data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -99,21 +119,35 @@ void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException
                 .build();
 
         final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
-        try {
-            s3ObjectReadTimer.recordCallable((Callable<Void>) () -> {
-                doParseObject(s3ObjectReference, getObjectRequest, bufferAccumulator);
-
-                return null;
-            });
-        } catch (final IOException | RuntimeException e) {
-            throw e;
-        } catch (final Exception e) {
-            // doParseObject does not throw Exception, only IOException or RuntimeException. But, Callable has Exception as a checked
-            // exception on the interface. This catch block thus should not be reached, but in case it is, wrap it.
-            throw new RuntimeException(e);
-        }
-
-        s3ObjectsSucceededCounter.increment();
+        //Issue#1971-Start
+		if (null != s3SourceConfig.getS3SelectOptions()) {
```

Comment:
In general, conditions like this are not desirable. A better approach would be to use a [strategy](https://en.wikipedia.org/wiki/Strategy_pattern). The `S3ObjectWorker` would receive an interface for how to get objects from S3. It won't know anything about either S3 Select or the Codecs.

Then there would be two strategies:
1. GetObject / codec strategy (the existing behavior)
2. S3 Select strategy